gcp_variant_transforms/libs/bigquery_vcf_schema.py

```diff
@@ -203,22 +210,57 @@ def generate_schema_from_header_fields(header_fields, variant_merger=None,
   return schema
 
 
-def get_row_from_variant(variant, split_alternate_allele_info_fields=True):
-  """Returns a BigQuery row according to the schema from the given variant.
+def get_rows_from_variant(variant, split_alternate_allele_info_fields=True):
+  """Yields BigQuery rows according to the schema from the given variant.
+
+  There is a 10MB limit for each BigQuqery row, which can exceed by having
+  a large number of calls. This method may split up a row into multiple rows if
+  it exceeds 10MB.
 
   Args:
     variant (``Variant``): Variant to process.
     split_alternate_allele_info_fields (bool): If true, all INFO fields with
       `Number=A` (i.e. one value for each alternate allele) will be stored under
       the `alternate_bases` record. If false, they will be stored with the rest
       of the INFO fields.
-  Returns:
-    A dict representing BigQuery row from the given variant.
+  Yields:
+    A dict representing a BigQuery row from the given variant. The row may have
+    a subset of the calls if it exceeds the maximum allowed BigQuery row size.
   Raises:
     ValueError: If variant data is inconsistent or invalid.
   """
   # TODO: Add error checking here for cases where the schema defined
   # by the headers does not match actual records.
+  base_row = _get_base_row_from_variant(
+      variant, split_alternate_allele_info_fields)
+  base_row_size_in_bytes = len(json.dumps(base_row))
```

Comment:
It would be good to abstract the length calculation to another method like `_get_row_size(row)`.

Reply:
Done. Good idea!